### PR TITLE
Rhcloud 19331 add pool type flag

### PR DESCRIPTION
**Pull request by maknop**
Added **--pool-type** flag and updated reservation template. Just wanted to ensure I was headed in the correct direction.

### CODE REVIEW

File: bonfire/bonfire.py [hunk 1]
```diff
@@ -208,6 +208,13 @@ def _validate_reservation_duration(ctx, param, value):
         help="Duration of the reservation",
         callback=_validate_reservation_duration,
     ),
+    click.option(
+        "--pool-type",
+        type=str,
```
Comment (bsquizz) [inline bonfire/bonfire.py:213]: If there's only a limited set of valid choices ('default' or 'minimal') -- check out the click 'choice' type: https://click.palletsprojects.com/en/8.1.x/options/#choice-options

File: bonfire/resources/reservation-template.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ objects:
   spec:
     duration: ${DURATION}
     requester: ${REQUESTER}
+    pool-type: ${POOLTYPE}
```
Comment (bsquizz) [inline bonfire/resources/reservation-template.yaml:16]: I suspect we are going to ask that the key name here be renamed to `poolType` over in https://github.com/RedHatInsights/ephemeral-namespace-operator/pull/65/files#diff-14d05dc348d738e3ff5d3d6a0ca563b6616adb212c02c12b6628be117d81e081R30 -- so you'll probably want to update it here too :)

**Review by bsquizz — CHANGES_REQUESTED** [review]

File: bonfire/resources/reservation-template.yaml (hunk 2 above)
Comment (maknop) [reply bonfire/resources/reservation-template.yaml:16]: Good point. Going to shorten it to just "pool" instead.

File: bonfire/bonfire.py (hunk 1 above)
Comment (maknop) [reply bonfire/bonfire.py]: Cool, just added this.

**Pull request merged**